akka-stream/src/main/scala/akka/stream/impl/fusing/ActorGraphInterpreter.scala

```diff
@@ -511,9 +530,23 @@ private[stream] class ActorGraphInterpreter(_initial: GraphInterpreterShell) ext
         false
     }
 
+  private def adjustEventLimit(shell: GraphInterpreterShell): Unit =
+    if (effectiveEventLimit < shell.shellEventLimit) effectiveEventLimit = shell.shellEventLimit
```

Comment:
This looks wrong: if a single shell in there has a limit of 1, then the whole actor will use this limit.